src/Client.php

```diff
@@ -353,13 +358,6 @@ private function applyOptions(RequestInterface $request, array &$options)
             unset($options['query']);
         }
 
-        if (isset($options['json'])) {
```

Comment:
I moved this up to be consistent with multipart and form_params in setting the body option.